llvm/utils/TableGen/IntrinsicEmitter.cpp

```diff
@@ -273,11 +273,12 @@ using TypeSigTy = SmallVector<unsigned char>;
 /// Computes type signature of the intrinsic \p Int.
 static TypeSigTy ComputeTypeSignature(const CodeGenIntrinsic &Int) {
   TypeSigTy TypeSig;
-  if (const auto *R = Int.TheDef->getValue("TypeSig")) {
-    for (const auto *a : cast<ListInit>(R->getValue())->getValues()) {
-      for (const auto *b : cast<ListInit>(a)->getValues())
-        TypeSig.emplace_back(cast<IntInit>(b)->getValue());
-    }
+  const Record *TypeInfo = Int.TheDef->getValueAsDef("TypeInfo");
+  const ListInit *OuterList = TypeInfo->getValueAsListInit("TypeSig");
```

Comment:
This is dropping the null check that was here before

Reply:
Yeah, the null check is not necessary anymore. It was needed to for unit test that had they own Intrinsic class definition, but now all of them have moved to use the Intrinsics.td file, so we can assume the Intrinsic class will have that field.

Reply:
It's better to not rely on properties that happen to hold in the predefined .td files

Reply:
That I don't understand. Isn't a particular emitter tightly coupled with the corresponding TableGen record structure? For example, code in `CodeGenIntrinsic.cpp` assumes that all Intrinsic records have fields TargetPrefix and LLVMName. In the same vein, this code assumes that it has a `TypeInfo` field.

Reply:
Yes, but tablegen shouldn't be crashy garbage on arbitrary inputs. If it's not literally baked into the code, and is input data like the predefined .td files, it needs to be verified

Reply:
Fair enough. Though in this case, I'd then amend the check to CodeGenInrinsic constructor to check this. Let me do that.

Reply:
Actually, looks like we are covered here. Record::getValueAsDef() does this check and will `PrintFatalError` is the field does not exist.